Scripts/Python/grsnMainWallPython.py

```diff
@@ -106,9 +113,16 @@ def OnSDLNotify(self,VARname,SDLname,playerID,tag):
                     break
                 southWall.value[blocker].runAttachedResponder(kBlockerOn)
         elif(nState == sState == kSelectCount and ageSDL["NumBlockers"][0] == 0):
-            for i in range(0,171):
-                northWall.value[i].runAttachedResponder(kBlockerOff)
-                southWall.value[i].runAttachedResponder(kBlockerOff)
+            if self.oldNorthWall:
+                for blocker in self.oldNorthWall:
+                    if(blocker == -1):
+                        break
+                    northWall.value[blocker].runAttachedResponder(kBlockerOff)
+            if self.oldSouthWall:
+                for blocker in self.oldSouthWall:
+                    if(blocker == -1):
+                        break
+                    southWall.value[blocker].runAttachedResponder(kBlockerOff)
```

Comment:
```suggestion
            for blocker in self.oldNorthWall:
                if blocker == -1:
                    break
                northWall.value[blocker].runAttachedResponder(kBlockerOff)
            for blocker in self.oldSouthWall:
                if blocker == -1:
                    break
                southWall.value[blocker].runAttachedResponder(kBlockerOff)
```

There shouldn't be any need to test the truthyness of the old wall sequences. An loop over an empty sequence does nothing. Also, we may want to consider changing the
```python
for blocker in foo:
    if blocker == -1:
       break
     # do work here
     pass
```
anti-patter to something more pythonic, such as [`itertools.takewhile`](https://docs.python.org/3/library/itertools.html#itertools.takewhile):
```python
for blocker in itertools.takewhile(lambda x: x!= -1, foo):
    # do work here
    pass
```